src/scripts/_fill/_fill.gml

```diff
@@ -0,0 +1,38 @@
+/// @func _fill(array, value [, start, end])
+/// @desc Fills elements of array with value from start up to, but not including, end.
+/// @param {Array} array The array to fill
+/// @param {*} value The value with which to fill elements of array
+/// @param {Integer} optionalStart The start index
+/// @param {Integer} optionalEnd The end index
+/// @returns {Array} The filled array
+/// @note This method mutates array.
+/*
+@example
+var arr = [0, 1, 2, 3];
+_fill(arr, 4, 1, 3);
+// => [0, 4, 4, 3];
+
+var arr = [0, 1, 2, 3];
+_fill(arr, 0);
+// => [0, 0, 0, 0];
+*/
+
+var i;
+var arr;
+var _end;
```

Comment:
Would be good to rename this. It's not unlikely that in the future there would be a script called `_end`.